### PR TITLE
Add missing PHPDoc tags

### PR DESCRIPTION
**Pull request by thiemowmde**
I'm focusing on cases that are not obvious when reading the method signature.

### CODE REVIEW

File: src/Patcher/MapPatcher.php [hunk 1]
```diff
@@ -89,6 +89,13 @@ private function applyOperation( &$base, $key, $diffOp ) {
 		}
 	}
 
+	/**
+	 * @param array &$base
+	 * @param string $key
```
Comment (JeroenDeDauw) [inline src/Patcher/MapPatcher.php]: key can be int and float as well if I'm not mistaken

File: src/Patcher/MapPatcher.php [hunk 2]
```diff
@@ -98,6 +105,13 @@ private function applyDiffOpAdd( &$base, $key, DiffOpAdd $diffOp ) {
 		$base[$key] = $diffOp->getNewValue();
 	}
 
+	/**
+	 * @param array &$base
+	 * @param string $key
```
Comment (JeroenDeDauw) [inline src/Patcher/MapPatcher.php]: key can be int and float as well if I'm not mistaken

File: src/Patcher/MapPatcher.php [hunk 3]
```diff
@@ -112,6 +126,13 @@ private function applyDiffOpRemove( &$base, $key, DiffOpRemove $diffOp ) {
 		unset( $base[$key] );
 	}
 
+	/**
+	 * @param array &$base
+	 * @param string $key
```
Comment (JeroenDeDauw) [inline src/Patcher/MapPatcher.php]: key can be int and float as well if I'm not mistaken

File: src/Patcher/MapPatcher.php [hunk 4]
```diff
@@ -126,6 +147,13 @@ private function applyDiffOpChange( &$base, $key, DiffOpChange $diffOp ) {
 		$base[$key] = $diffOp->getNewValue();
 	}
 
+	/**
+	 * @param array &$base
+	 * @param string $key
```
Comment (JeroenDeDauw) [inline src/Patcher/MapPatcher.php]: key can be int and float as well if I'm not mistaken

File: src/Patcher/MapPatcher.php [hunk 5]
```diff
@@ -139,11 +167,24 @@ private function applyDiff( &$base, $key, Diff $diffOp ) {
 		$base[$key] = $this->patchMapOrList( $base[$key], $diffOp );
 	}
 
+	/**
+	 * @param array &$base
+	 * @param string $key
```
Comment (JeroenDeDauw) [inline src/Patcher/MapPatcher.php]: key can be int and float as well if I'm not mistaken

File: src/Patcher/MapPatcher.php (hunk 1 above)
Comment (thiemowmde) [inline src/Patcher/MapPatcher.php]: Documented as `int|string` in https://github.com/wmde/Diff/blob/master/src/DiffOp/Diff/Diff.php#L409. Will fix.

**Pull request merged**